app/src/main/java/org/mozilla/fenix/ext/NavController.kt

```diff
@@ -34,3 +35,14 @@ fun NavController.navigateSafe(
         this.navigate(directions)
     }
 }
+
+/**
+ * Checks if the Fragment with a [fragmentClassName] is on top of the back queue.
+ */
+@SuppressLint("RestrictedApi")
+fun NavController.hasTopDestination(fragmentClassName: String): Boolean {
+    return this.backQueue.lastOrNull()?.destination?.displayName?.contains(
+        fragmentClassName,
+        true,
+    ) == true
+}
```

Comment:
this looks like a good approach. Another thought I had is that we could simply have a `fun NavController.getTopOfBackstackName()` and then we compare it in `onKeyUp()`. I don't lean a particular way, so I'll defer that final decision to you.

Reply:
I kind of do not like having to check nullable booleans with `==true` so I prefer having it just in the extension function instead of replicating in many condition checks (possibly multiple conditions like in the current patch).